consensus/spos/bls/subroundSignature.go

```diff
@@ -177,13 +186,13 @@ func (sr *subroundSignature) receivedSignature(_ context.Context, cnsDta *consen
 		return false
 	}
 
-	currentMultiSigner, err := sr.MultiSignerContainer().GetMultiSigner(sr.Header.GetEpoch())
+	err = sr.SignatureHandler().SetMultiSignerByEpoch(sr.Header.GetEpoch())
 	if err != nil {
-		log.Error("receivedSignature.GetMultiSigner", "error", err.Error())
+		log.Debug("receivedSignature.SetMultiSignerByEpoch", "error", err.Error())
 		return false
 	}
 
-	err = currentMultiSigner.VerifySignatureShare([]byte(node), sr.GetData(), cnsDta.SignatureShare)
+	err = sr.SignatureHandler().VerifySignatureShare(uint16(index), cnsDta.SignatureShare, sr.GetData())
```

Comment:
similar to my comment above, add epoch as parameter to VerifySignatureShare and remove the SetMultiSignerByEpoch above.

Reply:
added epoch parameter